benchmark/submit.py

```diff
@@ -0,0 +1,129 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+"""
+Submit benchmark results to a server.
+"""
+
+from __future__ import print_function
+
+import json
+import sys
+
+from twisted.internet import endpoints
+from twisted.internet.defer import succeed
+from twisted.internet.task import react
+from twisted.python import usage
+from twisted.web import client, http
+from twisted.web.http_headers import Headers
+from twisted.web.iweb import IBodyProducer
+
+from zope.interface import implementer
+
+
+@implementer(IBodyProducer)
+class StringProducer(object):
+    def __init__(self, body):
+        self.body = body
+        self.length = len(body)
+
+    def startProducing(self, consumer):
+        consumer.write(self.body)
+        return succeed(None)
+
+    def pauseProducing(self):
```

Comment:
:grey_question: nit: resumeProducing is also required in order to implement IBodyProducer.

This suggests you might want to make a unit tests and just run something from zope.interface.verify on this class.  Might be overkill for this utility though, I don't know.